src/components/Editor/QueryEditor.jsx

```diff
@@ -0,0 +1,50 @@
+import {required, SimpleForm, TextInput} from "react-admin";
+import config from "../../config";
+import {useEffect, useState} from "react";
+
+function QueryEditor(props) {
+
+    let {newQuery} = props
+
+    const [queryText, setQueryText] = useState()
+
+    let query;
+    if (!newQuery) {
+        query = config.queries.filter(
+            (query) => query.id === props.id
+        )[0];
+    }
+
+    useEffect(() => {
+        if (query) {
+            fetch(`${config.queryFolder}${query.queryLocation}`)
+                .then((response) =>
+                response.text())
+                .then((text) =>
+                setQueryText(text));
+        }
+    }, [])
+
+
+    const onSubmit = (data) => {
+        if (newQuery) {
+            // Create new query
+        } else {
+            // Edit existing query
+        }
+    }
+
+    return (
+        <SimpleForm onSubmit={onSubmit}>
+            <TextInput source="id" name={"Query ID"} validate={[required()]} defaultValue={query ? query.id : undefined}/>
+            <TextInput source="name" name={"Name"} validate={[required()]} defaultValue={query ? query.name : undefined}/>
+            <TextInput source="description" name={"Description"} defaultValue={query ? query.description : undefined}/>
+            <TextInput source="query" multiline={true} name={"Query"} defaultValue={queryText}/>
```

Comment:
This should be some sort of rich text input, allowing for a cleaner, multiline editing of the query text